[docs] Adds the token doc (#14857)
MystenLabs/sui: docs/content/standards/closed-loop-token.mdx

---
title: Closed Loop Token
---

import BalanceTokenCoin from '@site/static/img/balance-token-coin.png';

Closed Loop Token allows its creator to limit applications where the token is used, set up custom policies for transfers, spending and conversions. It is defined in the `sui::token` module and is a part of the Sui Framework.

## Background & Use Cases

> The standard Coin implementation on Sui is an example of an open-loop system - Coins are free-flowing, "wrappable", and the best real world analogy would be "cash" - hardly regulated and can be freely used and passed.

Reviewer: Link to https://docs.sui.io/concepts/object-ownership/wrapped on "wrappable"?

> The standard Coin implementation on Sui is an example of an open-loop system - Coins are free-flowing, [wrappable](/concepts/object-ownership/wrapped), [freely transferable](/concepts/dynamic-fields/transfers/custom-rules#the-store-ability-and-transfer-rules) and can be stored in any application. The best real world analogy would be "cash" - hardly regulated and can be freely used and passed.

Reviewer: More a comment for @ronny-mysten (and I think this is already recorded somewhere), but this reminds me that the `transfer/custom-rules` docs should not be under "dynamic fields".

Reviewer: It may be easier to follow if you start this section with the analogy that tokens are like bank accounts, and then follow it up with the example of `Coin` being like cash, because the main topic here is what tokens are, so it can be confusing for people to understand where the story is going if the first paragraph is about what tokens are not.

> However, some applications require constraining the scope of the token to a specific purpose. For example, a token that can only be used for a specific service, or it can be used only by an authorized account, or it can be used freely while account is not blocked. A real-world analogy would be a "bank account" - regulated, controlled by the bank and comply with certain rules and policies.

Reviewer: ```suggestion
However, some applications require constraining the scope of the token to a specific purpose. For example, a token that can only be used for a specific service, or that can be used only by an authorized account, or one whose accounts can be blocked. A real-world analogy would be a "bank account" - regulated, controlled by the bank and compliant with certain rules and policies.
```

> However, some applications require constraining the scope of the token to a specific purpose. For example, a token that can only be used for a specific service, or that can be used only by an authorized account, or one whose accounts can be blocked. A real-world analogy would be a "bank account" - regulated, controlled by the bank and compliant with certain rules and policies.

Reviewer 2: I have to say that the bank account reference does not resonate with me. But it's all good. Just mentioning in case you want to remove it. I think the 3 examples below are enough and good for me
